bundledatasrc.go

```diff
@@ -245,13 +247,18 @@ func parseBundleParts(r io.Reader) ([]*BundleDataPart, error) {
 	return parts, nil
 }
 
+// userFriendlyUnmarshalErrors provides user friendly versions of the
+// yaml unmarshall errors for a bundle.
+// Lines such as: "line 1: field name not found in type charm.ApplicationSpec"
+// are converted to: "line 1: unrecognized field name"
+// The actual errors for developers can be found with Trace level logging
+// enabled.
 func userFriendlyUnmarshalErrors(err error) error {
 	logger.Tracef("developer friendly error message: \n%s", err.Error())
 	friendlyText := err.Error()
-	friendlyText = strings.ReplaceAll(friendlyText, "type charm.ApplicationSpec", "applications")
```

Comment:
Just to make sure I understand this change:
Is this where the 'not found in applications' came from? eg it originally said 'unmarshal errors: field foo not found in type charm.ApplicationSpec'.
I think what you've done is definitely better. I wonder if we would rather: `" unrecognized field: "` because that gives us a slightly clearer: 
```
  line 5: unrecognized field: constrain
```
I would normally just quote it with:
```
  line 5: unrecognized field "constrain"
```

But I feel like some sort of character to distinguish what part of the sentence is the field name is useful.